engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -114,6 +114,23 @@ public void processRecord(final TypedRecord<ProcessInstanceMigrationRecord> comm
       return;
     }
 
+    final DeployedProcess sourceProcessDefinition =
```

Comment:
:x: We should also do a null-check for the `sourceProcessDefinition` (like [here](https://github.com/camunda/zeebe/pull/15519/files#diff-7388b163a2f978814a720c3ee81b65fd415dac06abf2aefa99881eacaab0b787R109-R115)) since the user may not be authorized to handle the data from the given tenant.

Reply:
@koevskinikola Is it possible that process instance's `tenantId` can be different than that instance's process definition's `tenantId`?

```java
final DeployedProcess sourceProcessDefinition =
        processState.getProcessByKeyAndTenant(
            processInstance.getValue().getProcessDefinitionKey(),
            processInstance.getValue().getTenantId());
```
Here we are retrieving `tenantId` from the process instance itself.